dbmanager/dbmanager.go

```diff
@@ -0,0 +1,116 @@
+package dbmanager
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	_ "github.com/lib/pq" //db driver for postgres
+	"golang.org/x/crypto/bcrypt"
+
+)
+
+//////// QUERIES
+const (
+	userExistsQuery  = `SELECT uid from users WHERE email = $1`
+	getUserInfoQuery = `SELECT * from users WHERE email = $1`
+	getUserPassQuery = `SELECT password from users WHERE email = $1`
+	createUserQuery  = `INSERT INTO users(uid,email,password) VALUES (DEFAULT, $1 , $2);`
+)
+
+//////// ERRORS
+var ErrUserNonexistant = errors.New("User does not exist")
+var ErrUserExists = errors.New("User already exists")
+
+type DBManager struct {
+	dbUser, dbPass, dbName string
+	DB                     *sql.DB
+}
+
+func New(user, pass, name string) (DBManager, error) {
+	d := DBManager{dbUser: user, dbPass: pass, dbName: name}
+	err := d.connectToPSQL()
+	return d, err
+}
+
+// VerifyUserExists returns an error if the user does not exist
+func (db *DBManager) VerifyUserExists(email string) error {
+	rows, err := db.DB.Query(userExistsQuery, email)
+	if err != nil {
+		return err
+	}
+
+	if !rows.Next() {
+		return ErrUserNonexistant
+	}
+	return nil
+}
+
+func (db *DBManager) GetUserPassHash(email string) ([]byte, error) {
+	err := db.VerifyUserExists(email)
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := db.DB.Query(getUserPassQuery, email)
+	if err != nil {
+		return nil, err
+	}
+
+	rows.Next()
+	var hash []byte
+	rows.Scan(&hash)
+	return hash, nil
+}
+
+// Returns err if user is not added to DB
+func (db *DBManager) AddNewUser(email, password string) error {
+	err := db.VerifyUserExists(email)
```

Comment:
Same here - this query is not needed. Because of the uniqueness constrain on the `email` column, you cannot insert a new user with the same email.